src/main/java/ttt/Board.java

```diff
@@ -1,94 +1,116 @@
 package ttt;
 
 import java.util.Arrays;
+import java.util.function.Function;
 
 import static ttt.PlayerSymbol.*;
-import static ttt.PlayerSymbol.VACANT;
 
 public class Board {
-    private PlayerSymbol[] grid = new PlayerSymbol[BOARD_DIMENSION * BOARD_DIMENSION];
-
     protected static final int BOARD_DIMENSION = 3;
     private static final int NUMBER_OF_SLOTS = BOARD_DIMENSION * BOARD_DIMENSION;
+    private Cell[] grid = new Cell[BOARD_DIMENSION * BOARD_DIMENSION];
 
     public Board() {
-        for (int i = 0; i < NUMBER_OF_SLOTS; i++) {
-            grid[i] = VACANT;
+        for (int cellIndex = 0; cellIndex < NUMBER_OF_SLOTS; cellIndex++) {
+            grid[cellIndex] = new Cell(calculateOffsetFor(cellIndex), VACANT);
         }
     }
 
-    public Board(PlayerSymbol... grid) {
-        this.grid = grid;
+    public Board(PlayerSymbol... initialGridLayout) {
+        for (int cellIndex = 0; cellIndex < NUMBER_OF_SLOTS; cellIndex++) {
+            this.grid[cellIndex] = new Cell(calculateOffsetFor(cellIndex), initialGridLayout[cellIndex]);
+        }
+    }
```

Comment:
If you now take a closer look at `Board()` and `Board(PlayerSymbol... initialGridLayout)` you will see some code repetition there (the `for` loop). Is there a way to extracting the building process of the board to a common method?